lib/serverengine/supervisor.rb

```diff
@@ -95,18 +103,32 @@ def create_server(logger)
       @server = @create_server_proc.call(@load_config_proc, logger)
     end
 
+    def stop_internal(stop_graceful)
+      begin
+        _stop(stop_graceful)
+      rescue Errno::EPIPE
+        # nothing to do
+      ensure
+        if @command_pipe
+          @command_pipe.close rescue nil
```

Comment:
> why is it safe to ignore this error?

If the child process is already stopped, the exception (`Errno::EPIPE`) raised, as you know.
Such situation is often occurs.
For example, "auto restart in limited ratio" spec in `spec/supervisor_spec.rb` does so.
This is the same meaning in [`process_manager.rb`](https://github.com/fluent/serverengine/blob/master/lib/serverengine/process_manager.rb#L309).
But, now I feel that this is not correct place to rescue.